server/sum-aggregator.go

```diff
@@ -1,33 +1,54 @@
 package server
 
 import (
+	"sync"
+
 	"github.com/fluxio/metricd"
 	"github.com/fluxio/metricd/pb"
 )
 
 type sumAggregator struct {
-	name    string
-	labels  metricd.LabelSet
-	sum     float64
-	lastTs  int64
-	updated bool
+	name      string
+	labels    metricd.LabelSet
+	sum       float64
+	bucketSum float64
+	lastTs    int64
+	updated   bool
+	m         sync.Mutex
 }
 
 func (a *sumAggregator) AddValue(value float64, ts int64) {
+	a.m.Lock()
+	defer a.m.Unlock()
+
 	a.sum += value
+	a.bucketSum += value
 	a.lastTs = ts
 	a.updated = true
 }
 
 func (a *sumAggregator) GetAggregations() []*pb.Metric {
+	a.m.Lock()
+	defer a.m.Unlock()
+
 	if a.updated {
 		a.updated = false
-		return []*pb.Metric{{
-			Name:   a.name + "_sum",
-			Labels: a.labels,
-			Value:  &pb.Metric_DoubleValue{a.sum},
-			Ts:     a.lastTs,
-		}}
+		aggs := []*pb.Metric{
+			{
+				Name:   a.name + "_sum",
+				Labels: a.labels,
+				Value:  &pb.Metric_DoubleValue{a.sum},
+				Ts:     a.lastTs,
+			},
+			{
+				Name:   a.name,
```

Comment:
It looks to me like the previous aggregatorUnit implementations write derived metrics at a different name than the original name. I assume this is to distinguish aggregated metrics from the original, raw metric name. On the other hand, it also looks to me like the original metric name is never used directly, so we could just declare that this aggregatorUnit is allocated the raw metric name. Is this correct?

I think we should make the design of this mechanism explicit: aggregatorUnit implementations should have to declare, at least in their doc comments if nothing else, what metric names they output when applied.